lib/presentation/widget/upload_request_group/upload_request_group_widget.dart

```diff
@@ -0,0 +1,212 @@
+              ],
+            ),
+          ),
+        ));
+  }
+
+  Widget _buildSimpleUploadRequestListWidget(BuildContext context, List<UploadRequestGroup> uploadRequestList) {
+    return RefreshIndicator(
+        onRefresh: () async => _model.getUploadRequestCreatedStatus(),
+        child: uploadRequestList.isEmpty
+            ? _buildCreateUploadRequestsHere(context)
+            : ListView.builder(
+                itemCount: uploadRequestList.length,
+                itemBuilder: (context, index) => _buildSimpleUploadTile(uploadRequestList[index])));
+  }
+
+  Widget _buildActiveClosedListWidget(
+      BuildContext context, List<UploadRequestGroup> uploadRequestList) {
+    return RefreshIndicator(
+        onRefresh: () async => _model.getUploadRequestActiveClosedStatus(),
+        child: uploadRequestList.isEmpty
+            ? _buildCreateUploadRequestsHere(context)
+            : ListView.builder(
+                itemCount: uploadRequestList.length,
+                itemBuilder: (context, index) =>
+                    _buildActiveClosedUploadTile(uploadRequestList[index])));
+  }
+
+  ListTile _buildSimpleUploadTile(UploadRequestGroup request) {
+    return ListTile(
+      dense: true,
+      leading: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+        SvgPicture.asset(imagePath.icCollective, width: 20, height: 24, fit: BoxFit.fill)
+      ]),
+      title: Text(request.label,
+          style: TextStyle(fontSize: 14, color: AppColor.uploadRequestLabelsColor)),
+      subtitle: Text(
+          AppLocalizations.of(context)
+              .activated_date(request.activationDate.getMMMddyyyyFormatString()),
+          style: TextStyle(fontSize: 14, color: AppColor.uploadFileFileSizeTextColor)),
+      trailing: IconButton(
+          icon: SvgPicture.asset(
+            imagePath.icContextMenu,
+            width: 24,
+            height: 24,
+            fit: BoxFit.fill,
+          ),
+          onPressed: () => null),
+      onTap: () => _model.goToUploadRequests(request),
+    );
+  }
+
+  ListTile _buildActiveClosedUploadTile(UploadRequestGroup request) {
+    return ListTile(
+      dense: true,
+      leading: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+        SvgPicture.asset(imagePath.icCollective, width: 20, height: 24, fit: BoxFit.fill)
+      ]),
+      title: Text(request.label,
+          style: TextStyle(fontSize: 14, color: AppColor.uploadRequestLabelsColor)),
+      subtitle: _buildActiveClosedSubtitleWidget(request.status),
+      trailing: IconButton(
+          icon: SvgPicture.asset(
+            imagePath.icContextMenu,
+            width: 24,
+            height: 24,
+            fit: BoxFit.fill,
+          ),
+          onPressed: () => null),
+        onTap: () => _model.goToUploadRequests(request),
+    );
+  }
+
+  Widget _buildActiveClosedSubtitleWidget(UploadRequestStatus status) {
```

Comment:
It's duplicated LOC for these cases. They are the only differences in the Decoration color. Pls write a common function for them.

Reply:
Text are also differents between status so I think separate them is a good thing

Reply:
I mean this code block:

```
return Row(children: [
  Container(
    margin: EdgeInsets.only(right: 4),
    width: 7.0,
    height: 7.0,
    decoration: BoxDecoration(
      color: AppColor.primaryColor,
      shape: BoxShape.circle,
    ),
  ),
  Text(AppLocalizations.of(context).active,
      style: TextStyle(fontSize: 14, color: AppColor.uploadFileFileSizeTextColor))
]);
```

Then only passing the different things via params